### PR TITLE
Add curvefit trainer (fix minor issue, update test data)

### PR DESCRIPTION
**Pull request by sunya-ch**
This PR is mainly to introduce a new trainer class `curvefit` with two trainers (LogarithmicRegressionTrainer and LogisticRegressionTrainer) in correspondence to https://github.com/sustainable-computing-io/kepler-model-server/issues/214.

In addition to that, this PR also fix some minor issues including
- broken link to tekton page
- process_accelerator_feature function removal (should not replace features of trainer there)
-`get_query_results` function reuse in model_tester.py
- unused import removal
- update .gitignore to allow updating prom_output 
- unused tests/prom_reponse* removal 
- add missing `abs-train-pipelinerun.yaml`
- change test data to stressng data instead of coremark

Signed-off-by: Sunyanan Choochotkaew <sunyanan.choochotkaew1@ibm.com>

### CODE REVIEW

**Comment by sunya-ch** [comment]
Example results of curvefit trainer on SPEC data:
1. LogisticRegressionTrainer 
Screenshot 2024-01-16 at 14 56 32

2. LogarithmicRegressionTrainer
Screenshot 2024-01-16 at 14 55 24

**Comment by rootfs** [comment]
this looks very cool!

**Review by rootfs — APPROVED** [review]

File: src/train/trainer/curvefit.py [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+        weight_dict = dict()
+
+        for component, model in self.node_models[node_type].items():
+            scaler = self.node_scalers[node_type]
+            weight_dict[component] = {
+                "All_Weights": {
+                    "Categorical_Variables": dict(),
+                    "Numerical_Variables": {self.features[i]: 
+                                            {"scale": scaler.scale_[i]} for i in range(len(self.features))},
+                    "CurveFit_Weights": list(model.popt)
```
Comment (rootfs) [inline src/train/trainer/curvefit.py:132]: does the weights contain info about which model to use? Would it be possible we can unit test the model with mock data in both python and golang? I am going to write a golang library to implement the logistic and logaritham

Comment (sunya-ch) [reply src/train/trainer/curvefit.py:132]: Yes, it can be extracted and use it directly also on golang.

**Pull request merged**